### DISCUSSION TITLE
How Security Onion Handles correlation use cases?

### DISCUSSION THREAD

**Discussion by Tedus007** [Security-Onion-Solutions/securityonion#5573] [Unsupported Versions]
Dear Team,

Need to understand the way we can build correlation use cases based on logs injected to SO.

I understand we use Playbook, Suricate and Zeek for our alerting mechanism.

But is there a way to build a correlation use case based on the logs collected from different source, windows, firewall, exchange,...etc.

Can we add time as a factor and number of generated logs to look for anomaly log patterns?

**Comment by dougburks** [comment]
Depending on what you're trying to do, you might take a look at ElastAlert:
https://docs.securityonion.net/en/2.3/elastalert.html

**Reply by Tedus007** [reply]
I'm trying to run simple correlation rules on windows logs.

Currently I'm using beats to ship windows logs to SO.

Is there is away to create a rule to trigger when successive failed login occurs during specific period of time via playbook or we need to use Wazuh for the correlation part and logs should be coming from Wazuh

**Reply by dougburks** [reply]
ElastAlert is fairly flexible. I'd encourage you to check out their documentation:
https://elastalert.readthedocs.io/en/latest/
